_episodes/24.md

```diff
@@ -0,0 +1,413 @@
+   - Node - The physical machine or VM running a kubelet and container runtime. 
+   - Cluster - a set of nodes connected to the same control plane node.
+   - Pod - single instance of an application, the smallest object in kubernetes.
+
+   <p align="center">
+    <img align="center" width="75%" height="100%" src="/assets/episode/24/components-of-kubernetes.svg"/><br/>
+    Source: https://kubernetes.io/docs/concepts/overview/components/
+   </p>
+
+   Kubernetes control plane components:
+     - API server that kubelets connect with and is the front end for 
+     users and administrators of the cluster.
+     - etcd keystore is a distributed store containing all data used to manage 
+     the cluster
+     - Scheduler that distributes work across nodes and assigns newly created 
+     containers to nodes
+     - Controllers that are the brain behind orchestration and monitors for 
+     nodes going down etc...
+     
+   Kubernetes worker nodes components:
+     - container runtime - is the underlying runtime used to manage containers 
+     - kubelet - agent that checks the health and manages the pods running on the node based on the desired state provided in the PodSpec
+     - kube-proxy - network proxy that maintains network rules applied to nodes and allows network access between Pods in a cluster
+ 
+   You can scale up multiple PODS on a single node until the node has no more 
+   resources, at which time a new node will need to be added and pod instances will
+   be distributed between the nodes. 
+
+1. **So how does this relate to Trino?** 
+     - Why not just use docker compose? Docker-compose is a tool that creates
+     multiple containers on a single host, where kubernetes is a platform that
+     allows complex distributed workloads and self healing applications across
+     nodes.
+     - Out of the box for any Kubernetes cluster can do these key things for Trino.
+        - Simple scale up and down (manually tell k8s to startup or kill).
+        - Kubernetes supports failover, meaning that your workers will self heal if they die.
+     - Advanced jobs that could exist but not currently in open source.
+        - Auto-scaling via the [Horizontal Pod Autoscaler](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/) 
+        and custom metrics.
+        - Graceful Shutdowns are hooks that you can add into your cluster that
+         <p align="center">
+            <img align="center" width="75%" height="100%" src="/assets/episode/24/kubernetes-shutdown.svg"/><br/>
+            Source: https://learnk8s.io/graceful-shutdown
+         </p>
+         <p align="center">
+            <img align="center" width="75%" height="100%" src="/assets/episode/24/graceful-shutdown.svg"/><br/>
+            Source: https://learnk8s.io/graceful-shutdown
+         </p>
+1. **What the heck are helm charts then?**
+According to the helm site, Helm is the best way to find, share, and use 
```

Comment:
Rejig this paragraph logically

- Helm is package manager for K8s
- Removes the need for managing lots of k8s related yaml files
- Best way to deploy apps to k8s
- Charts are available for many different applications
- Helm chart for Trino